common/src/main/java/com/wynntils/screens/base/widgets/TextInputBoxWidget.java

```diff
@@ -199,15 +200,13 @@ private Pair<String, Integer> getRenderedText(float maxTextWidth) {
 
         StringBuilder builder = new StringBuilder();
 
-        // First append to the left of the cursor
-        int stringPosition = cursorPosition - 1;
-        while (font.width(builder.toString()) < maxTextWidth && stringPosition >= 0) {
-            builder.append(textBoxInput.charAt(stringPosition));
-
+        int stringPosition = cursorPosition;
+        while (font.width(builder.toString()) < maxTextWidth && stringPosition > 0) {
             stringPosition--;
+            builder.append(textBoxInput.charAt(stringPosition));
         }
 
-        final int startingAt = Math.max(stringPosition, 0); // If we went too far, start at the beginning
+        final int startingAt = stringPosition;
```

Comment:
Does this fix anything or did you just refactor this?

Reply:
This is a fix. Assuming the text was larger than the maximum text width by 2 chars, the behavior was:
- Decrement `stringPosition`
- while `stringPosition`>= 0
  - Add char at `stringPosition`
  - Decrement `stringPosition`

It means that for the latest iteration, we get `stringPosition` offseted by `-1` instead of the actual index of the last char added. That's why I simply switched the two instructions in the loop and removed the initial decrementation

Reply:
Sounds good. Okay.